### PR TITLE
Junsulime

### PR DESCRIPTION
**Pull request by JunsuLime**

### CODE REVIEW

**Review by wrlawodms — COMMENTED** [review]
흠..

File: Recursive_ExhaustiveSearch/BOARDCOVER_lim.py [hunk 1]
```diff
@@ -0,0 +1,149 @@
+import sys
```
Comment (seohui) [inline Recursive_ExhaustiveSearch/BOARDCOVER_lim.py]: sys

File: Recursive_ExhaustiveSearch/PICNIC_lim.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
+import itertools
+
```
Comment (seohui) [inline Recursive_ExhaustiveSearch/PICNIC_lim.py]: 피크닉~

File: Recursive_ExhaustiveSearch/PICNIC_lim.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+	current_base = 0
+	for idx, s in enumerate(map(int, input().split())):
+		if idx % 2 == 0:
+			current_base = s
+		else:
+			# if a is b's friend, b is a's friend too.
+			friend_map[current_base].add(s)
+			friend_map[s].add(current_base)
+	
+	def matching():
```
Comment (seohui) [inline Recursive_ExhaustiveSearch/PICNIC_lim.py]: matching

File: Recursive_ExhaustiveSearch/PICNIC_lim.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+			return match_count
+	
+		# start matching work from 0
+		return __matching_internal(0, set())
+	
+	return matching()
+
+
+for _ in range(test_case):
```
Comment (seohui) [inline Recursive_ExhaustiveSearch/PICNIC_lim.py]: aaa

**Review by JunsuLime — COMMENTED** [review]
Add comment about code

**Review by JunsuLime — COMMENTED** [review]
Another comment

**Pull request merged**